crates/sui-benchmark/src/workloads/adversarial.rs

```diff
@@ -214,3 +234,53 @@ impl Workload<dyn Payload> for AdversarialWorkload {
             .collect()
     }
 }
+
+struct AdversarialPayloadArgs {
+    pub fn_name: String,
+    pub args: Vec<CallArg>,
+}
+
+fn get_payload_args(
+    payload_type: AdversarialPayloadType,
+    protocol_config: &ProtocolConfig,
+) -> AdversarialPayloadArgs {
+    match payload_type {
+        AdversarialPayloadType::ObjectsSize => AdversarialPayloadArgs {
+            fn_name: "create_shared_objects".to_owned(),
+            args: [
+                // TODO: Raise this. Using a smaller value here as full value locks up local machine
+                (NUM_OBJECTS / 10).into(),
+                // TODO: Raise this. Using a smaller value here as full value locks up local machine
+                (protocol_config.max_move_object_size() / 10).into(),
+            ]
+            .to_vec(),
+        },
+        AdversarialPayloadType::EventSize => AdversarialPayloadArgs {
+            fn_name: "emit_events".to_owned(),
+            args: [
+                // TODO: Raise this. Using a smaller value here as full value locks up local machine
+                (protocol_config.max_num_event_emit() / 10).into(),
+                // TODO: Raise this. Using a smaller value here as full value errs
+                (protocol_config.max_event_emit_size() / 3).into(),
+            ]
+            .to_vec(),
+        },
+        AdversarialPayloadType::DynamicFieldsCount => AdversarialPayloadArgs {
```

Comment:
For this test, what I had in mind was not adding dynamic fields (which basically looks the same as object creation from the runtime's perspective), but reading a bunch of dynamic fields added by a previous tx (which is a good stress test because it's the only time the Move VM needs to go to the DB).

For this, I'd suggest:
- adding a module initializer to `adversarial` that creates one object, then adds a large number of dynamic fields to it
- adding this object ID as a field in`TestPayload`
- make this logic call a function that reads N dynamic fields from that object

Reply:
Besides dynamic field reads, I think another good on would be trying to read a lot of shared objects at once. For that I think the recipe is similar: create the shared objects in a module initializer, remember them in the payload, use them in the logic

Reply:
Good suggestions. I did the first but was only able to add 33 dynamic fields before running out of gas. Will improve this in a subsequent PR.
Will also add the shared object read suggestion. In the meantime, I want to unblock this PR.

Reply:
Rather than publishing a separate package, I would suggest calling `add_n_dynamic_fields` one time, inside `Workload::init`. That will maximize the gas you have available for initialization, and will be simpler than adding a separate package.